app/src/debugger.lisp

```diff
@@ -0,0 +1,304 @@
+;;;; src/debugger.lisp
+;;;;
+;;;; Author: Juan M. Bello-Rivas
+
+(defpackage #:qvm-app.debugger
+  (:use #:common-lisp #:qvm-app)
+  (:export #:debugger)
+  (:import-from #:alexandria #:assoc-value #:once-only))
+
+(in-package #:qvm-app.debugger)
+
+(defparameter *threshold* (* 1e3 double-float-epsilon) "Smallest amplitude that can be printed.")
+
+(defparameter *qvm* nil "Current QVM.")
+
+(defparameter *display* nil "Current display setting.")
+
+(defparameter *source-filename* nil "Currently loaded source file.")
+
+(defparameter *breakpoints* nil "List of breakpoints.")
+
+(defparameter *prompt* "(qvm-debugger) ")
+
+(defun error-unless-qvm-exists ()
+  (unless *qvm*
+    (error "No program has been loaded.")))
+
+(defstruct command
+  (name nil :type string :read-only t)
+  (function nil :type function :read-only t)
+  (documentation nil :type string :read-only t))
+
+(defparameter *commands* nil
+  "Association list of available commands.")
+
+(defmacro define-command (name lambda-list &body body)
+  "Define a new command function named %NAME (note that the % character is prepended to NAME) with the specified LAMBDA-LIST and BODY.
+
+Documentation strings are mandatory in debugger commands and they should be formatted as follows:
+
+1. First line containing usage information.
+2. Empty line.
+3. Longer description in a line by itself.
```

Comment:
What about evaluating `command-description` here in `define-command` and store it on a description slot of the `command` struct? Or perhaps just checking that the docstring conforms to this specification here. What I see below is that for `help` the description is used explicitly, but not so much for the other commands.

Reply:
I agree 100% that that is the better way to go. However, I think the current implementation is good enough for this PR. The descriptions are used in `help` and also in `help COMMAND`.